File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/animations/RespawnAnimationTest.java
package com.bonepl.chromaleague.hud.animations;

import com.bonepl.chromaleague.GameStateMocks;
import com.bonepl.chromaleague.hud.AnimationTester;
import com.bonepl.chromaleague.rest.activeplayer.ChampionStats;
import org.junit.jupiter.api.Test;

import static org.mockito.Mockito.when;

class RespawnAnimationTest {

    @SuppressWarnings("JUnitTestMethodWithNoAssertions")
    @Test
    void playRespawnAnimation() {
        // given
        GameStateMocks gameStateMocks = new GameStateMocks();

        final ChampionStats championStats = gameStateMocks.championStats();
        when(championStats.resourceMax()).thenReturn(100.0);
        when(championStats.resourceValue()).thenReturn(100.0);
        when(championStats.currentHealth()).thenReturn(100.0);
        when(championStats.maxHealth()).thenReturn(100.0);

        when(gameStateMocks.player().getChampionName()).thenReturn("Morgana");

        // then
        new AnimationTester().testAnimation(new RespawnAnimation());
    }
}

File: test/hud/animations/RespawnAnimation.test.ts
import {describe, test} from 'vitest';
import {AnimationTester} from '../../AnimationTester.js';
import {GameStateMocks} from '../../GameStateMocks.js';
import {RespawnAnimation} from '../../../src/hud/animations/RespawnAnimation.js';

/**
 * Integration test for respawn animation on actual hardware.
 */
describe('RespawnAnimation Integration', () => {
    test('playRespawnAnimation', async () => {
        // given
        const mocks = new GameStateMocks();

        const championStats = mocks.championStats();
        championStats.resourceMax = 100;
        championStats.resourceValue = 100;
        championStats.currentHealth = 100;
        championStats.maxHealth = 100;

        mocks.playerList();

        // then
        await new AnimationTester().testAnimation(new RespawnAnimation());
    }, 30000);
});
